### ISSUE TITLE
No mouse input on Steam when display is not at 0,0

### ISSUE THREAD

**Issue by dgbuckley** [ifreund/river#391]
To reproduce:
- Have any connected displays not at 0,0 (My 3 are at 1530,0 / 0,700 / 1080,1080)
- Open steam and try to click anywhere

If I move a monitor to 0,0 then input works on that monitor. This was also tested on a single output laptop with the same results.

**Comment by Leon-Plickat** [comment]
I think this is an XWayland issue, in which case we can't really do anything to fix it.

**Comment by dgbuckley** [comment]
I don't have this issue on sway which I assume I would if this was an XWayland issue.

**Comment by ifreund** [comment]
The fact that this only happens with steam seems very weird. I wonder what make steam different from every other X11 client I've tested, which all work fine.

**Comment by RobertMueller2** [comment]
I've encountered a very similar issue a while ago with hikari. The description reads identical to me, but right now I can't confirm with a test (external screen broken). I'll do so when I get a chance.

BTW, I've not seen the problem with Plasma (Wayland), so I also don't really think it's Xwayland.

**Comment by RobertMueller2** [comment]
> I've encountered a very similar issue a while ago with hikari. The description reads identical to me, but right now I can't confirm with a test (external screen broken). I'll do so when I get a chance.

I've just tried this with river and wlr-randr. Of course I can't say if it has the same root cause, but it looks identical to the issue I've  observed with hikari earlier.

**Comment by dedguy21** [comment]
I have the same issue with Steam. I have to open it with steams Big Picture mode to get around the issue.

**Comment by ifreund** [comment]
So steam's big picture mode works properly but steam itself doesn't? I hate X11...

**Comment by ifreund** [comment]
This seems to be fixed by a3fdb294b311216bec388f97a52148ff8e1155cd, don't ask me why :D

**Comment by dgbuckley** [comment]
I'm not sure why, but even after building from latest I still have the issue. It seems like a position issue; when I open steam with my output at position 1080,1080 I can't click anything but if my output is at position 540,540 I can click in the bottom right and from the bottom right open top left dropdowns.

![steam_issue](https://user-images.githubusercontent.com/44792985/134982328-d439b34c-2d01-49f5-9429-8ebbc28a0509.png)

**Comment by ifreund** [comment]
Indeed, it seems that a3fdb294b311216bec388f97a52148ff8e1155cd has not affected this issue and my testing was insufficient.

What I've found is that if steam is started on an output at 0,0 mouse input will work properly, even if steam is moved to an output not at 0,0.

However, if started on an output not at 0,0, mouse input will never work properly even if moved to an output at 0,0. 

This is pretty weird.

**Comment by tysongeorge** [comment]
I seem to have this issue as well on my second monitor (-1080,0); whether it's steam, firefox, discord (flatpak) or other apps -- starting on my main monitor (at 0,0 where the apps function as expected) and moving them over doesn't allow me to use mouse input either. 

On the second monitor mouse input via tmux works fine though, same with OBS (flatpak) and Xournal++.

**Comment by ifreund** [comment]
@tysongeorge what river version are you using? (run `river -version`)

**Comment by tysongeorge** [comment]
I'm currently on 0.2.4

**Comment by ifreund** [comment]
@tysongeorge It looks like this is an Xwayland bug caused by your negative coordinates and X11's inability to deal well with them: https://gitlab.freedesktop.org/xorg/xserver/-/issues/899.

Perhaps river, wlroots, or Xwayland will eventually find some reasonable way to fix this but for now I'd recommend sticking to positive output coordinates for your Xwayland usage.

**Comment by tysongeorge** [comment]
Ah, that's interesting; thanks nonetheless!
